### DISCUSSION TITLE
'Missing Data' error during sync

### DISCUSSION THREAD

**Discussion by jencastrodoesstuff** [digidem/mapeo-mobile#795] [Ideas]
After unsucsesfull attept to sync different devices with MD

**Describe the bug**
Error Messages after sync but data appears to have transferred.

**Can you do the same thing twice?**
difficult to repeat the sequence of events

1. After unsuccessful attempt to sync different devices with MD...
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-092305_Mapeo QA](https://user-images.githubusercontent.com/33264095/86799654-2df9b180-c040-11ea-871f-5e804f922e0f.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-091845_Mapeo QA](https://user-images.githubusercontent.com/33264095/86799789-4f5a9d80-c040-11ea-90cf-45f2f9cb2ece.jpg)

2. Check observation list for synced data
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-093119_Mapeo QA](https://user-images.githubusercontent.com/33264095/86799895-71ecb680-c040-11ea-9ecf-2459a48aa02a.jpg)
S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-093109_Mapeo QA](https://user-images.githubusercontent.com/33264095/86799944-82049600-c040-11ea-93a2-fba703fb4157.jpg)

3. Create new observation
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-093411_Mapeo](https://user-images.githubusercontent.com/33264095/86800008-98125680-c040-11ea-9042-fb529c7bb9c7.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-093520_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800059-a2cceb80-c040-11ea-865d-60dcec213494.jpg)

4. Return to sync screens and hit sync
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-095827_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800772-5f26b180-c041-11ea-9f11-c7e089a928e8.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-095829_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800743-5930d080-c041-11ea-847d-a259c505a5c7.jpg)

5.  and watch progress
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-095851_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800804-677eec80-c041-11ea-99c7-0094b2e91903.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-095835_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800847-74034500-c041-11ea-8292-fdcdd88b8e15.jpg)
![Screenshot_20200707-095911_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800862-7796cc00-c041-11ea-89dc-0fbdfdc565c7.jpg)

6. See error in v5
S10 - Mm v2.1.0-RC.781-QA
No error but no confirmation 
![Screenshot_20200707-095907_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800818-6b127380-c041-11ea-8379-f8c3222f5a53.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-095911_Mapeo QA](https://user-images.githubusercontent.com/33264095/86800953-8a110580-c041-11ea-8af7-577d7aa88d37.jpg)

7. Check synced data - it appears to have worked
S10 - Mm v2.1.0-RC.781-QA
![Screenshot_20200707-095929_Mapeo QA](https://user-images.githubusercontent.com/33264095/86808003-71f0b480-c048-11ea-9223-6c33102df6d4.jpg)

S8 - Mm v5.0.0 beta 1 
![Screenshot_20200707-095929_Mapeo QA](https://user-images.githubusercontent.com/33264095/86808025-774dff00-c048-11ea-8388-2bcbbcc45cb8.jpg)

**Expected behavior**
A confirmation that sync completed successfully

**Comment by okdistribute** [comment]
This will be fixed in the next version, got a bandaid fix here 
(https://github.com/digidem/mapeo-core/pull/97) and @noffle is working on another approach

**Comment by ghost** [comment]
> @noffle is working on another approach

fyi for v9, not v8

**Comment by okdistribute** [comment]
This should be fixed in latest v5 RC, please re-open if you continue to see this issue when there is definitely no data missing :)

**Comment by jencastrodoesstuff** [comment]
Reopening this as the error message issue. Since doe appear to be happening for with both Mm and Md but the error messages are still appearing.
Here are side by side simultaneous videos

[VIDEO: S8 (start 3 seconds after s10 video)](https://drive.google.com/file/d/1bN94KmuRCLTmzyGSW9NMnETi2iPqavbH/view?usp=sharing)

 
[VIDEO: S10 ](https://drive.google.com/file/d/1IZ0o_HVpsUl3wVAbgleo_EThqiCs84-9/view?usp=sharing )

**Comment by okdistribute** [comment]
Seems like this could actually be a case of missing data, what is the desired behavior in that case? (before, the sync would never end.. but that isn't right.)

**Comment by jencastrodoesstuff** [comment]
[VIDEO S10 syncing with Desktop after some edits to observations](https://drive.google.com/file/d/1IqTGTDdk6iSAmbsftjlfJYYAe4iHLg6x/view?usp=sharing)

**Comment by jencastrodoesstuff** [comment]
Desired behaviour is identifying the observation that has missing data... at some point, if a device ID/name were included then it may become possible to find the observation source to solve the problem or optionally delete the incomplete data.... info about how to resolve the issue is generally the best solution.

Aliya's input would be valuable here.
